application/dash_application/views/bamqc_gbovertime.py

```diff
@@ -26,24 +21,23 @@
 bamqc[COL_TOTAL_BASES] = bamqc[col.TotalReads] * bamqc[col.AverageReadLength]
 bamqc[COL_RUN_DATE] = pd.to_datetime(bamqc[col.Run].str[0:6], format="%y%m%d")
 
-bamqc_lib = [
-    {"label": x, "value": x} for x in bamqc[col.SequencingType].unique()
-]
+bamqc_lib = [{"label": x, "value": x} for x in bamqc[col.SequencingType].unique()]
 bamqc_lib.append({"label": "All", "value": "All"})
 
 layout = html.Div(
     [
-        core.Dropdown(id=ids['lib'], options=bamqc_lib, value="All"),
-        core.Graph(id=ids['month_plot']),
-        core.Graph(id=ids['3month_plot']),
-        core.Graph(id=ids['cum_plot']),
+        core.Dropdown(id=ids["lib"], options=bamqc_lib, value="All"),
+        core.Graph(id=ids["month_plot"]),
+        core.Graph(id=ids["3month_plot"]),
+        core.Graph(id=ids["cum_plot"]),
     ]
 )
 
+
 def init_callbacks(dash_app):
     @dash_app.callback(
-        Output(ids['month_plot'], "figure"), 
-        [Input(ids['lib'], "value")])
+        Output(ids["month_plot"], "figure"), [Input(ids["lib"], "value")]
```

Comment:
Is there any way to tweak the code formatter's behaviour? I think having the Input array to the same line as Output reduces the readability of callbacks, and it seems inconsistent about when it does this, too

Reply:
Black's philosophical underpinning is that things like this simply aren't configurable. I tried to jiujitsu the formatter by adding `#` + newline between input/output items but it still reformatted them onto the same line (with one `#` at the end of the line, sigh). I do agree that it makes the inputs and outputs more difficult to read, which is concerning given how central they are to dashi. I'd like to look into other automatic formatters that don't sacrifice readability. @slazicoicr did you investigate any other formatters when you were looking at Black?

Reply:
Not too deeply, as Black worked well for gsiqcetl out of the box. There is:
* autopep8 
* yapf

There is a roundup at https://medium.com/3yourmind/auto-formatters-for-python-8925065f9505 which does state that yapf is way more configurable than black.

Reply:
Ok. I'm going to take a look at the formatters and see if we can't find something that maintains readability.